### PR TITLE
PUPIL-678 feat(lesson-listing): style lesson listing page

### PR DESCRIPTION
**Pull request by benprotheroe**
## Description

Music year: [2007](https://www.youtube.com/watch?v=PsO6ZnUZI0g)
Music year: [1979](https://www.youtube.com/watch?v=fiD39jo5Yo4)

Add style to Lesson lIsting pages

- creating a view file to create the UI
- use oak-components to build UI

### Acceptance criteria

- [x]  page matches designs
- [x]  passing tests

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2412--oak-web-application.netlify.thenational.academy/pupils/beta/programmes/maths-secondary-year-11-higher-l/units/further-algebraic-fractions-2758/lessons
2. Click on lesson or back button
3. You should see all the available legacy lessons; note that the lesson listing is not yet navigable from the actual lesson pages

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

**Comment by messmike** [comment]
@benprotheroe, thanks few things:

- Can we capitalise 'lessons' e.g: 
Screenshot 2024-05-08 at 12 38 16 pm

- The back button label currently says 'Change units', can we remove the 's' at the end?

- The background illustration svg looks quite large and I can only see a part of it on desktop - can it be resized? E.g: 
- Screenshot 2024-05-08 at 12 44 00 pm

- I'm assuming that the correct nav item will be highlighted at the top of the page when that part is picked up? Same with the footer link - if so, ignore this part.

**Review by messmike — APPROVED** [review]
giphy

**Comment by RBrownsett** [comment]
@benprotheroe looking good. A couple of points from an accessibility point of view:

**Info Component:**

image

Using a screen reader the text in the info component is not read out (its quite tricky to find the actual text in dev tools). 

Would it be possible to amend this so there is an aria-expanded attribute that toggles to true/false and a aria-controls attribute that points to the element containing the help text? By the looks of it tooltip needs to be amended in the dom order so it follows the button. Reading about tooltips here https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role#keyboard_interactions  

Whether we go down the tool-tip route or pivot to a toggle-tip, which sounds more like the functionality that we have created (tool-tips tend to be activated on hover or focus)

https://carbondesignsystem.com/components/toggletip/usage/


**Lesson List Section**

More of a query of whether this section would be best placed in an ordered list opposed to a collection of divs? Again gives a screen reader the ability to inform the users that there is list of lessons

image

**Comment by RBrownsett** [comment]
@benprotheroe might not be possible but is there anything we can do about the white space when a lesson title spans more than one line (lesson 1 below)? Doesn't look too bad on desktop but on tablet is quite noticeable  

image

**Comment by kimon-satan** [comment]
> Would it be possible to amend this so there is an aria-expanded attribute that toggles to true/false and a aria-controls attribute that points to the element containing the help text? By the looks of it tooltip needs to be amended in the dom order so it follows the button. Reading about tooltips here https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role#keyboard_interactions

@benprotheroe could you make a backlog ticket for this ?

**Comment by kimon-satan** [comment]
> **Lesson List Section**
> 
> More of a query of whether this section would be best placed in an ordered list opposed to a collection of divs? Again gives a screen reader the ability to inform the users that there is list of lessons

Will be easier if we can use an aria role for this so that we don't have to reset the OL and LI elements .
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/list_role
@benprotheroe  can you make a ticket for this ?

**Comment by kimon-satan** [comment]
> @benprotheroe might not be possible but is there anything we can do about the white space when a lesson title spans more than one line (lesson 1 below)? Doesn't look too bad on desktop but on tablet is quite noticeable
> 
> image

I think we need to consult with @messmike about this one . Looking here the issue is that the gap between the two elements is a bit large
https://components.thenational.academy/?path=/docs/components-organisms-pupil-oakpupiljourneylistitem--docs
image

**Review by kimon-satan — APPROVED** [review]
Just one line of code to change for me, but I don't need to review again.

File: src/components/PupilViews/PupilLessonListing/PupilLessonListing.view.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    props;
+  const {
+    yearDescription,
+    subject,
+    subjectSlug,
+    tierDescription,
+    examboardDescription,
+    phaseSlug,
+  } = programmeFields;
+  const phase = phaseSlug === "primary" ? "primary" : "secondary";
```
[the remark this answers is not in the document]
Comment (benprotheroe) [reply src/components/PupilViews/PupilLessonListing/PupilLessonListing.view.tsx:38]: in owa or oak-components?

**Comment by messmike** [comment]
Currently, the space between the title and icon/label is set to `space-between-s`; it could help to reduce the amount of whitespace to use s`pace-between-sssx` if that is an easy update to make. 
Screenshot 2024-05-08 at 4 33 13 pm

**Comment by messmike** [comment]
> Currently, the space between the title and icon/label is set to `space-between-s`; it could help to reduce the amount of whitespace to use s`pace-between-sssx` if that is an easy update to make.

**Review by RBrownsett — APPROVED** [review]
Happy that the accessibility items will be picked up on the relevant components in additional tickets by @benprotheroe. Tested across various browsers and screen sizes with no issues. Content matches designs.

**Pull request merged**